src/service-override/views.ts

```diff
@@ -197,7 +203,7 @@ type Label = string | {
 interface EditorPanelOption {
   readonly id: string
   name: string
-  renderBody (container: HTMLElement): IDisposable
+  renderBody (container: HTMLElement): BodyRenderer
```

Comment:
I'm not really happy with extending IDisposable here

an alternative that is used a lot inside VSCode itself is to provide a DisposableStore as the last parameter

Reply:
The risk I see with that is that it's easy to forget for the consumer. If they forget to handle that parameter, then it won't get disposed. Now the type system will give an error if `dispose()` is not implemented on the returned value, making sure that things get cleaned up when we get rid of them.

Reply:
The dispose function being mandatory in on the `cons` column for me :smile: most of the time, you don't need it (for instance if you render a react component using portal?)

Reply:
In our React case we do need to handle dispose, because when dispose is called we need to remove the component from our list. Otherwise the component will try to render on an element that is not attached to the DOM.

Reply:
As an example, in our React case I actually realised that I forgot to properly dispose when I shared the code with you, because the dispose body was empty. I've since changed the code to this:

<img width="768" alt="image" src="https://github.com/CodinGame/monaco-vscode-api/assets/587016/215d7cb7-5067-4b5d-ab93-4d0e99fe96cf">

Reply:
As a last point, I think that this can feel like VSCode, with how they often use classes for UI. E.g. you could also pass this to it:

```ts
class MarkdownRenderer extends Disposable {
  constructor() { }
  
  setInput(input) {
    this.draw();
  }
  
  private draw() { /* ... */ }
}
```